### ISSUE TITLE
RuntimeError : about tf.Gradient() in eager mode

### ISSUE THREAD

**Issue by Johnson-yue** [tensorflow/tensorflow#23794]
**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04):   Linux Ubuntu 16.04
- TensorFlow installed from (source or binary):  pip installed
- TensorFlow version (use command below):   1.10.0
- Python version: 3.5
- CUDA/cuDNN version: cuda-9.0, cudnn-7.2
- GPU model and memory:   GTX 1080 ti - 11G

**Describe the current behavior**
I have a huge model with large parameter, before train code, cost GPU memory is 7859MB, and  my local memory unused is 234M free(total is 32G)
![image](https://user-images.githubusercontent.com/10268274/48605840-dca73500-e9b8-11e8-84ac-8a7a8bb617ba.png)
![image](https://user-images.githubusercontent.com/10268274/48605858-ea5cba80-e9b8-11e8-92b9-3bc184829bba.png)

when I bug the code ,occur this warning:
```
2018-11-16 15:52:30.386249: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1484] Adding visible gpu devices: 0
2018-11-16 15:52:30.571530: I tensorflow/core/common_runtime/gpu/gpu_device.cc:965] Device interconnect StreamExecutor with strength 1 edge matrix:
2018-11-16 15:52:30.571559: I tensorflow/core/common_runtime/gpu/gpu_device.cc:971]      0
2018-11-16 15:52:30.571565: I tensorflow/core/common_runtime/gpu/gpu_device.cc:984] 0:   N
2018-11-16 15:52:30.571742: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1097] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 9772 MB memory) -> physical GPU (device: 0, name: GeForce GTX 1080 Ti, pci bus id: 0000:01:00.0, compute capability: 6.1)
2018-11-16 15:52:34.410871: W tensorflow/core/framework/allocator.cc:108] Allocation of 423405000 exceeds 10% of system memory.
2018-11-16 15:52:34.602313: W tensorflow/core/framework/allocator.cc:108] Allocation of 372480000 exceeds 10% of system memory.

```
I will update my discriminator network so I compute gradient first and in eager mode, I used tf.GradientTape() for it.
But Exception here: I already used tf.GradientTape() for gradients but, it also warning tf.gradients is not supported when eager execution is enabled.  I did not use tf.gradient anywhere!!!!
![image](https://user-images.githubusercontent.com/10268274/48606113-a9b17100-e9b9-11e8-856b-e88a5bbed34e.png)

and I test demo code like line:1508-1511 in pictures, it worked well. But line:1557 it did not work with same code!!!!  I am sure that I did not use tf.gradients() in my code!!

**Describe the expected behavior**

It should be work 

**Other info / logs**
Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached.
![image](https://user-images.githubusercontent.com/10268274/48606265-30664e00-e9ba-11e8-9688-553d27c0ab45.png)

**Comment by akshaym** [comment]
I have a fix for this which should be merged soon.
